pkg/lib/queueinformer/jitter.go

```diff
@@ -1,32 +1,42 @@
 package queueinformer
 
 import (
+	"math"
 	"math/rand"
 	"time"
 )
 
 const DefaultResyncPeriod = 15 * time.Minute
 
+type float64er interface {
+	Float64() float64
+}
+
+type realFloat64er struct{}
+
+func (realFloat64er) Float64() float64 {
+	return rand.Float64()
+}
+
 // ResyncWithJitter takes a resync interval and adds jitter within a percent difference.
 // factor is a value between 0 and 1 indicating the amount of jitter
 // a factor of 0.2 and a period of 10m will have a range of 8 to 12 minutes (20%)
 func ResyncWithJitter(resyncPeriod time.Duration, factor float64) func() time.Duration {
+	return resyncWithJitter(resyncPeriod, factor, realFloat64er{})
+}
+
+func resyncWithJitter(period time.Duration, factor float64, rand float64er) func() time.Duration {
 	return func() time.Duration {
-		if factor < 0.0 || factor > 1.0 {
-			return resyncPeriod
-		}
-		if resyncPeriod < 0.0 {
+		if period < 0.0 {
 			return DefaultResyncPeriod
 		}
-
-		// if we would wrap around, return resyncPeriod
-		if time.Duration((1+factor)*resyncPeriod.Minutes())*time.Minute < 0.0 {
-			return resyncPeriod
+		if factor < 0.0 || factor > 1.0 {
+			return period
 		}
-
-		min := resyncPeriod.Minutes() * (1 - factor)
-		max := resyncPeriod.Minutes() * (1 + factor)
-
-		return time.Duration(min)*time.Minute + time.Duration(rand.Float64()*(max-min))*time.Minute
+		if period > math.MaxInt64/2 { // 1281023h53m38.427387903s
```

Comment:
nit: could we move this `period` comparison up a few lines right after the previous `period` comparison? Makes it a little easier to read.